dirtylarry/dirtylarry.lua

```diff
@@ -50,6 +64,10 @@ function dirtylarry.checkbox(self, node, action_id, action, value)
 	local node_bg = gui.get_node(node .. "/larrycheckbox")
 	local node_label = gui.get_node(node .. "/larrylabel")	
 
+	if not dirtylarry.is_enabled(self, node_bg) then
+		return
```

Comment:
`return checked` here so it always returns the input unchanged even if it's disabled